### DISCUSSION TITLE
minio mac check in GCM failed

### DISCUSSION THREAD

**Discussion by withyoubytime** [minio/minio#20658] [Q&A]
👋 Hi everyone!
I used the official Docker image to deploy the MinIO server.
I'm encountering an issue when accessing a MinIO server deployed on an ARM64 architecture using the MinIO Admin Client. The same client works fine when accessing a MinIO server deployed on an AMD64 server. However, when targeting the ARM64 server, I receive the following error: org.bouncycastle.crypto.InvalidCipherTextException: mac check in GCM failed. 
MinIO server version is:
./ minio -v
minio version RELEASE.2023-03-20T20-16-18Z (commit-id=05444a0f6af8389b9bb85280fc31337c556d4300)
Runtime: go1.19.7 linux/arm64
License: GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>
Copyright: 2015-2023 MinIO, Inc.
Minio Admin Client version:8.5.12
My test code is as follows:
`    @Test
    public void createServiceAccount(){
        String endpoint="http://xxx:port";
        String accessKey = "ak";
        String secretKey = "sk";
        // policyMap
        Map<String, Object> policyMap = new HashMap<>();

        // Version
        policyMap.put("Version", "2012-10-17");

        // Statement
        List<Map<String, Object>> statements = new ArrayList<>();

        // Statement的HashMap
        Map<String, Object> statement = new HashMap<>();
        statement.put("Effect", "Allow");

        // Action是一个List，所以我们创建一个List来存储它
        List<String> actions = new ArrayList<>();
        actions.add("s3:*");
        statement.put("Action", actions);

        // Resource也是一个List，所以我们创建一个List来存储它
        List<String> resources = new ArrayList<>();
        resources.add("arn:aws:s3:::*");
        statement.put("Resource", resources);

        // 将Statement添加到Statements列表中
        statements.add(statement);

        // 将Statements列表添加到顶层HashMap中
        policyMap.put("Statement", statements);
        MinioAdminClient adminClient =
                MinioAdminClient.builder().endpoint(endpoint).credentials(accessKey, secretKey).build();
        try {
            adminClient.addServiceAccount("3j6q268rjloyxw92","16BuRBVLU78UYMiWEIT8pZVhHw32hTtq",
                    null,policyMap,null,null,null);
        } catch (NoSuchAlgorithmException | InvalidKeyException | IOException | InvalidCipherTextException e) {
            throw new RuntimeException(e);
        }
    }`
The behavior when accessing a MinIO server deployed on ARM64 is as follows:
![arm64](https://github.com/user-attachments/assets/1a9c473e-7970-4e7d-a33d-4928dc3d5ae4)
The behavior when accessing a MinIO server of the same version deployed on AMD64 is as follows:
Successfully running.
![amd64](https://github.com/user-attachments/assets/1cab06d5-a026-40cb-a3ee-b8f25351df4f)

**Comment by marktheunissen** [comment]
> RELEASE.2023-03-20T20-16-18Z

Please upgrade to the latest release, and repeat your test. If it still fails we can take a look.

**Reply by withyoubytime** [reply]
The issue persists even though I am using the latest version.

bash-5.1# minio -v
minio version RELEASE.2024-11-07T00-52-20Z (commit-id=cefc43e4daa4cbb490ef6726ea374e26a93eb85e)
Runtime: go1.23.3 linux/arm64
License: GNU AGPLv3 - https://www.gnu.org/licenses/agpl-3.0.html
Copyright: 2015-2024 MinIO, Inc.
When connecting to the same Minio server, I tested another method in the Minio Admin Client called serverInfo(), and it is functioning correctly.

**Reply by marktheunissen** [reply]
Ok I think it's already being tracked here: https://github.com/minio/minio-java/issues/1295 please follow that issue

**Reply by withyoubytime** [reply]
> Ok I think it's already being tracked here: [minio/minio-java#1295](https://github.com/minio/minio-java/issues/1295) please follow that issue

Thank you for your support.
